### DISCUSSION TITLE
Enabling the use of ? on Option<> and Result<> By Handling Multiple Error Types

### DISCUSSION THREAD

**Discussion by WarrenN1** [rwf2/Rocket#2549] [Questions]
Basically, I am using uuid::Uuid as well as the Scylla Rust Driver. The problem is that the Scylla Rust Driver throws its own error types of QueryError and FromRowError and the Uuid::parse_str throws uuid::error. My question is that in my route handler function I would effectively like the ability to simply do:
```
let uuid  = Uuid::parse_str(session_cookie.value())?;
let rows = session.query("SELECT * FROM test.table", (&session_id,)).await?;
```
The problem is I do not know how to approach this in rocket as I am new.

**Comment by Mike-Neto** [comment]
One approach would be to map the errors into a custom error type that you create so that all of them map to the type you declare on the route. In the following example I'm returning the status directly but you can apply this mapping strategy for anything.

```
#[macro_use]
extern crate rocket;

#[get("/<name>/<age>")]
fn hello(name: &str, age: u8) -> Result<String, rocket::http::Status> {
    let data =
        std::fs::read_to_string("example.txt").map_err(|_err| rocket::http::Status::NotFound)?;
    let bytes = vec![age];
    let char_age = String::from_utf8(bytes).map_err(|_err| rocket::http::Status::BadRequest)?;
    Ok(format!(
        "Hello, {} year old named {}! with data {} and char age: {}",
        age, name, data, char_age
    ))
}

#[launch]
fn rocket() -> _ {
    rocket::build().mount("/hello", routes![hello])
}
```
